### PR TITLE
List GCE instances and store in DB

### PR DESCRIPTION
**Pull request by cfconrad**
This is the first step to get GCE support.
It works to list instances and store them into the DB.

### CODE REVIEW

File: webui/ocw/lib/gce.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        if GCE.__instance is None:
+            GCE.__instance = object.__new__(cls)
+            GCE.__instance.__credentials = GCECredential()
+        return GCE.__instance
+
+    def compute_client(self):
+        if self.__credentials.isExpired():
+            self.__credentials.renew()
+            self.__compute_clinet = None
+        self.project = self.__credentials.getInfo()['project_id']
```
Comment (asmorodskyi) [inline webui/ocw/lib/gce.py]: why project is not listed in the very beginning of the class and don't have '__' before name ?

File: webui/ocw/lib/vault.py [hunk 2]
```diff
@@ -113,16 +114,31 @@ def getCredentials(self):
...
-        cred_file = json.loads(base64.b64decode(self.getData('private_key_data')))
+        if 'errors' in creds:
+            raise Exception(",".join(creds['errors']))
+        cred_file = json.loads(base64.b64decode(creds['data']['private_key_data']).decode(encoding='UTF-8'))
         for k, v in cred_file.items():
             if k not in creds['data']:
                 creds['data'][k] = v
         return creds
+
+    def getInfo(self):
```
Comment (asmorodskyi) [inline webui/ocw/lib/vault.py]: let's name function according what they do
```suggestion
    def getPrivateKeyData(self):
```

File: webui/ocw/lib/gce.py (hunk 1 above)
Comment (cfconrad) [reply webui/ocw/lib/gce.py]: I think, python don't require it.
But I think it make sense. I will change it.

File: webui/ocw/lib/vault.py (hunk 2 above)
Comment (cfconrad) [reply webui/ocw/lib/vault.py]: ok

File: webui/ocw/lib/gce.py (hunk 1 above)
Comment (asmorodskyi) [reply webui/ocw/lib/gce.py]: yeah I know , just wanted to understand why you treat project field differently from others ?

Comment (cfconrad) [reply webui/ocw/lib/gce.py]: lazy python programmer :)

**Pull request merged**